### PR TITLE
refactor: split sendForm and sendConfirmation deep link operations

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary
We need to refactor deep link slightly before adding support for native tokens
Todo:
- [x] Add deep link simulator in the developer tools tab
- [x] Split send form and send confirmation form deep link
- [x] Add no back button prop to send confirmation form from deep link
- [x] Add disable expiration time changes for send confirmation deep link
- [x] Add disable gift storage deposit changes for send confirmation deep link

## Changelog

```
- Please write a list of changes
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

...

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/popups/SendConfirmationPopup.svelte [hunk 1]
```diff
@@ -175,9 +182,16 @@
...
+        {#if disableBack}
+            <Button classes="w-full" secondary onClick={onCancel} disabled={$isTransferring}>
+                {localize('actions.cancel')}
+            </Button>
+        {:else}
+            <Button classes="w-full" secondary onClick={onBack} disabled={$isTransferring}>
+                {localize('actions.back')}
+            </Button>
+        {/if}
+
```
Comment (maxwellmattryan) [inline packages/shared/components/popups/SendConfirmationPopup.svelte:194]: We can shorten this to:
```svelte
<Button classes="w-full" secondary onClick={disableBack ? onCancel : onBack} disabled={$isTransferring}>
    {localize(`actions.${disableBack ? 'cancel' : 'back'}`)}
</Button>
```

File: packages/shared/lib/core/wallet/interfaces/send-confirmation-parameters.interface.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { IAsset, Subject } from '@core/wallet'
+
+export interface ISendConfirmationParameters {
+    asset?: IAsset
+    amount?: string | undefined
+    unit?: string
+    recipient?: Subject
+    metadata?: string
+    tag?: string
+    giftStorageDeposit?: boolean
```
Comment (maxwellmattryan) [inline packages/shared/lib/core/wallet/interfaces/send-confirmation-parameters.interface.ts:10]: Does this not need the two other send parameters for disabling the gifting of the deposit, etc.? Might be related to the above comment about the different between "send operation parameters" and "send confirmation operation parameters"?

**Review by maxwellmattryan — COMMENTED** [review]
Looking really good, just have a few minor comments :+1:

File: packages/shared/locales/en.json [hunk 3]
```diff
@@ -1029,7 +1029,9 @@
         "unhideToken": "Unhide token",
         "burnToken": "Burn token",
         "faucetRequest": "Get {token} tokens",
-        "refreshTokenMetadata": "Refresh token metadata"
+        "refreshTokenMetadata": "Refresh token metadata",
+        "test": "Test",
+        "testDeepLink": "Test deep Link"
```
Comment (maxwellmattryan) [inline packages/shared/locales/en.json:1034]: This should be "Test deep link"

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]

**Comment by maxwellmattryan** [comment]
Some thoughts:
- We may want to disallow the user from deleting the prefix text of the URL, so that the wrong prefix can never be used. Or if the wrong prefix _is_ used then just display a warning notification against
- When deep links are disabled and the dev tries to use this functionality, we should either **1)** just disable / grey the button out in the first place or **2)** display a warning toast saying "Please enable deep links to use this functionality", maybe even with an action on the notification / alert to do that. I think 2 is a friendlier option since it's clearer what the dev needs to do in order to make the tool work.

**Comment by maxwellmattryan** [comment]
As I understand, deep link error handling is supposed to be improved in a near-future PR. The functionality works, and it's seriously amazing that we now have this and don't have to compile the app. Thanks for the new dev tool :pray:

File: packages/shared/components/popups/SendConfirmationPopup.svelte (hunk 1 above)
Comment (nicole-obrien) [reply packages/shared/components/popups/SendConfirmationPopup.svelte:194]: I think this is less readable, in my version you can clearly see the two different components to be displayed

File: packages/shared/lib/core/wallet/interfaces/send-confirmation-parameters.interface.ts (hunk 2 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/wallet/interfaces/send-confirmation-parameters.interface.ts:10]: Yeah added them

**Comment by nicole-obrien** [comment]
> Some thoughts:
> 
> * We may want to disallow the user from deleting the prefix text of the URL, so that the wrong prefix can never be used. Or if the wrong prefix _is_ used then just display a warning notification against
> * When deep links are disabled and the dev tries to use this functionality, we should either **1)** just disable / grey the button out in the first place or **2)** display a warning toast saying "Please enable deep links to use this functionality", maybe even with an action on the notification / alert to do that. I think 2 is a friendlier option since it's clearer what the dev needs to do in order to make the tool work.
> * I used this deep link (missing `s` in `rms`, `firefly-alpha://wallet/sendConfirmation?address=rm1qpgn2aelvx6qgxyvyyuvk5qrj6ps9mv7s5zlx5gygdzlgh7gpg3cwr536uv&amount=100&metadata=128347213&tag=soonaverse`) and IMO it should have displayed an error or warning, but instead it took me to the dashboard without doing or showing anything

Yeah we will improve error handling of deep links in a separate PR.

As for the deep links being enabled or not, this is a poor feature and just wrong. Thats because it doesnt register or deregister the deep link. But just turns of the handling of deep links and sends the user to the settings page, "technically" handling the deep link anyway. I think we should remove it and open a task to register and deregister the protocol with the os.

**Comment by maxwellmattryan** [comment]
> I think we should remove it and open a task to register and deregister the protocol with the os.

This makes sense :+1:

File: packages/shared/locales/en.json [hunk 4]
```diff
@@ -1526,7 +1530,7 @@
     "tabs": {
         "wallet": "Wallet",
         "staking": "Staking",
-        "developerTools": "Developer Tools"
+        "developerTools": "Developer"
```
Comment (maxwellmattryan) [inline packages/shared/locales/en.json:1533]: Should we also change `developerTools` to `developer` to match?

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
One minor comment / question

File: packages/shared/locales/ur.json [hunk 5]
```diff
@@ -1524,7 +1524,7 @@
     "tabs": {
         "wallet": "Wallet",
         "staking": "Staking",
-        "developerTools": "Developer Tools"
+
```
Comment (maxwellmattryan) [inline packages/shared/locales/ur.json:1527]: This is missing `"developer"`?

File: packages/shared/locales/vi.json [hunk 6]
```diff
@@ -1524,7 +1524,7 @@
     "tabs": {
         "wallet": "Wallet",
         "staking": "Staking",
-        "developerTools": "Developer Tools"
+
```
Comment (maxwellmattryan) [inline packages/shared/locales/vi.json:1527]: Same as above.

File: packages/shared/locales/zh-CN.json [hunk 7]
```diff
@@ -1524,7 +1524,7 @@
     "tabs": {
         "wallet": "钱包",
         "staking": "质押",
-        "developerTools": "Developer Tools"
+
```
Comment (maxwellmattryan) [inline packages/shared/locales/zh-CN.json:1527]: Same as above.

File: packages/shared/locales/zh-TW.json [hunk 8]
```diff
@@ -1524,7 +1524,7 @@
     "tabs": {
         "wallet": "錢包",
         "staking": "質押",
-        "developerTools": "Developer Tools"
+
```
Comment (maxwellmattryan) [inline packages/shared/locales/zh-TW.json:1527]: Same as above.

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Some potential mishaps with migrating some locale strings?

**Review by maxwellmattryan — APPROVED** [review]
LGTM :fire:

**Comment by adamunchained** [comment]
> Some thoughts:
> 
> * We may want to disallow the user from deleting the prefix text of the URL, so that the wrong prefix can never be used. Or if the wrong prefix _is_ used then just display a warning notification against
> * When deep links are disabled and the dev tries to use this functionality, we should either **1)** just disable / grey the button out in the first place or **2)** display a warning toast saying "Please enable deep links to use this functionality", maybe even with an action on the notification / alert to do that. I think 2 is a friendlier option since it's clearer what the dev needs to do in order to make the tool work.
> * I used this deep link (missing `s` in `rms`, `firefly-alpha://wallet/sendConfirmation?address=rm1qpgn2aelvx6qgxyvyyuvk5qrj6ps9mv7s5zlx5gygdzlgh7gpg3cwr536uv&amount=100&metadata=128347213&tag=soonaverse`) and IMO it should have displayed an error or warning, but instead it took me to the dashboard without doing or showing anything

just curious here, did Soonaverse generate deep-link without "s"?
